### PR TITLE
fix(react-searchbox): resolve ghost pointer events

### LINKED ISSUE #32926: [Bug]: SearchBox contentAfter is clickable when hidden
### Component

SearchBox

### Package version

9.54.17

### React version

-

### Environment

```shell
-
```

### Current Behavior

contentAfter is clickable although its not visible. It also will partially block anything if placed next to it.

image

### Expected Behavior

contentAfter shouldnt have pointer events or should not be clickable when not visible.

### Reproduction

https://react.fluentui.dev/?path=/docs/components-searchbox--docs

### Steps to reproduce

Go to storybook for Searchbox
Click in the empty space to the right of the search box
It will focus the searchbox

### Are you reporting an Accessibility issue?

None

### Suggested severity

Medium - Has workaround

### Products/sites affected

Clipchamp

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by carlamntn**
## New Behavior

see PR title

## Related Issue(s)

- Fixes #32926

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.148 MB`<br />`287.388 kB` | `1.148 MB`<br />`287.399 kB` | `9 B` <br />`11 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.14 kB`<br />`20.137 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `218.842 kB`<br />`63.382 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.447 kB`<br />`14.59 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `107.387 kB`<br />`35.758 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/0a71e8c7d271e6bad9a3878a084f6a40c5a93d1a'>0a71e8c7d271e6bad9a3878a084f6a40c5a93d1a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/Avatar-mount_base.html">613</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/Avatar-mount.html">628</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/Button-mount_base.html">307</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/Button-mount.html">305</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/Field-mount_base.html">1137</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/Field-mount.html">1135</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/FluentProvider-mount_base.html">724</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/FluentProvider-mount.html">722</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/FluentProviderWithTheme-mount_base.html">89</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/FluentProviderWithTheme-mount.html">87</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">36</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">37</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">85</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/MakeStyles-mount_base.html">883</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/MakeStyles-mount.html">849</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/Persona-mount_base.html">1783</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/Persona-mount.html">1727</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/SpinButton-mount_base.html">1351</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/SpinButton-mount.html">1379</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/SwatchPicker-mount_base.html">1683</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33005/perf-test-react-components/SwatchPicker-mount.html">1695</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts
Comment (fabricteam) [inline packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts:74]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=33005&commitId=bcf232110c2c46ae1911ce66e8a8095721b1f288&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|4| Changed|
|Avatar Converged.badgeMask.chromium.png|5| Changed|

</details>


  <details><summary><b>Drawer</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.overlay drawer full.chromium.png|1154| Changed|
|Drawer.Full Overlay RTL.chromium.png|1167| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by carlamntn** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts
Comment (tudorpopams) [inline packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts:77]: Can this be completely removed?

Comment (carlamntn) [reply packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts]: yes, it should be removed. Thanks!

File: packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts
Comment (Hotell) [inline packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts:80]: **question:** curious why using overflow instead of previously  used visibility: hidden ?

**Review by Hotell — APPROVED** [review]
LGTM

- lets resolve the comments from VR test tools
- once done, press the squash and merge button 

congrats on your 1st PR !

File: packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts
Comment (carlamntn) [reply packages/react-components/react-search/library/src/components/SearchBox/useSearchBoxStyles.styles.ts:80]: They are similar, overflow has the same behavior and is also accessibility compliant

**Pull request merged**
